### DISCUSSION TITLE
How to debug ImageGlass v9 on VS 2022?

### DISCUSSION THREAD

**Discussion by siblount** [d2phap/ImageGlass#1740] [Q&A] [answered]
All of the WebView2 components do not show in VS 2022, even in designer mode.
When I try to debug, I had to edit the FrmQuickSetup to skip past that form first. But things like the Settings, don't show up:

![image](https://github.com/d2phap/ImageGlass/assets/74916907/1e0eeb5f-4345-4ed2-9771-e214a0436e0a)

Of course, this works with the build you published. Is there anything else I need to get the WebView2 components to show?

**Comment by d2phap** [comment, marked as answer]
You need to 
- Open Terminal, navigate to the folder `\ImageGlass\Source\Components\ImageGlass.Settings\WebUI\`
- Run command `npm i` to install JavaScript dependencies
- Run command `npm start` to start watching and compiling js files
- Open VS 2022, click on menu Build > Rebuild Solution, it will copy all js files to the output folder
- Press F5 to start debugging the C# code, you should see the Web UI there.

To enable debugging Web UI:
- In `igconfig.json` file, set `EnableDebug` to `true` then launch IG
- Open any Web form, press <kbd>Ctrl+Shift+I</kbd> to open DevTool
